pkg/checks/base.go

```diff
@@ -56,6 +56,9 @@ type Check interface {
 	Schema() (*openapi3.SchemaRef, error)
 	// GetMetricCollectors allows the check to provide prometheus metric collectors
 	GetMetricCollectors() []prometheus.Collector
+	// RemoveLabelledMetrics allows the check to remove prometheus metric collectors
```

Comment:
This is not for rm the metric collectors. It's for removing metrics by target from available collectors

Reply:
Yes, the metrics struct in prometheus implements the collector interface, that's why I used the name collector... The GaugeVec, for example, is just a `Collector`, as everything else. The naming of the `GetMetricCollectors` is also not the best. :shrug:

I'll just change the naming in the comment. I'm still unsure on whether the interface is the proper place for metrics handling...